crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -292,6 +295,53 @@ impl BeaconState {
         )
     }
 
+    /// Check if ``indexed_attestation`` is not empty, has sorted and unique indices and has a valid
+    /// aggregate signature.
+    pub fn is_valid_indexed_attestation(&self, indexed_attestation: &IndexedAttestation) -> bool {
+        let indices: Vec<usize> = indexed_attestation
+            .attesting_indices
+            .iter()
+            .map(|&i| i as usize)
+            .collect();
+        // Verify indices are sorted and unique
+        if indices.is_empty() || !is_sorted_and_unique(&indices) {
+            return false;
+        }
+
+        // Collect public keys of attesting validators
+        let pubkeys: Vec<_> = indices
+            .iter()
+            .filter_map(|&i| self.validators.get(i).map(|v| v.pubkey.clone()))
+            .collect();
+
+        // Compute domain and signing root
+        let domain = self.get_domain(
+            DOMAIN_BEACON_ATTESTER,
+            Some(indexed_attestation.data.target.epoch),
+        );
+
+        let sig =
+            match blst::min_pk::Signature::from_bytes(&indexed_attestation.signature.signature) {
+                Ok(sig) => sig,
+                Err(_) => return false,
+            };
```

Comment:
```suggestion
        let sig =blst::min_pk::Signature::from_bytes(&indexed_attestation.signature.signature).map_err(|err| anyhow!(..))?;
```

if this fails don't return false it is a error which should be returned